torch/csrc/utils/tensor_new.cpp

```diff
@@ -220,6 +220,16 @@ Tensor internal_new_from_data(
   }
 
 #ifdef USE_NUMPY
+  if (PyObject_HasAttrString(data, "__cuda_array_interface__")) {
```

Comment:
Is there any reason to only enable this when we compile with `USE_NUMPY`? It seems to me that it should work even without Numpy.